x/staking/src/abci_handler.rs

```diff
@@ -53,9 +53,16 @@ impl<
     }
 
     fn init_genesis<DB: Database>(&self, ctx: &mut InitContext<'_, DB, SK>, genesis: GenesisState) {
-        self.keeper
-            .init_genesis(ctx, genesis)
-            .expect("Cannot perform genesis actions.")
+        // TODO
+        if let Err(err) = self.keeper.init_genesis(ctx, genesis) {
+            panic!(
+                "It is important to perform genesis actions.
+            During genesis staking module initialize pools and queues
+            using the parameters provided by the application.
+            Original error: {:?}",
+                err
```

Comment:
You shouldn't need to panic here. The keeper `init_genesis` method should be infallible.